components/dashboard/src/components/ConfirmationModal.tsx

```diff
@@ -6,89 +6,85 @@
 
 import Alert from "./Alert";
 import Modal, { ModalBody, ModalFooter, ModalHeader } from "./Modal";
-import { useRef, useEffect, ReactNode } from "react";
+import { FC, ReactNode, useCallback, useState } from "react";
 import { Button, ButtonProps } from "./Button";
 
-export default function ConfirmationModal(props: {
+type Props = {
     title?: string;
     areYouSureText?: string;
     children?: Entity | ReactNode;
     buttonText?: string;
     buttonDisabled?: boolean;
-    buttonLoading?: boolean;
     buttonType?: ButtonProps["type"];
     visible?: boolean;
     warningHead?: string;
     warningText?: string;
     footerAlert?: ReactNode;
     onClose: () => void;
-    onConfirm: () => void;
-}) {
-    const buttonType = props.buttonType || "danger";
-    const cancelButtonRef = useRef<HTMLButtonElement>(null);
+    onConfirm: () => void | Promise<void>;
+};
+export const ConfirmationModal: FC<Props> = ({
```

Comment:
Destructuring props here allows for easier default values, and avoids inefficiencies w/ passing full `props` object in a hook dep array.